Add multiple test platforms

csrakowski opened the pull request:
I have added .NET 4.6.1 and .NET 5.0 to the UnitTests project, and added .NET 5.0 to the build matrix.

csrakowski commented:
Hi @dvsekhvalnov,

Would it possible for you to have a look at my open PRs?

dvsekhvalnov commented:
Hi @csrakowski , sure thing. Sorry i missed notification about it.

What it is doing?

csrakowski commented:
Hi @dvsekhvalnov,
No worries; I remember from a few other tickets/PRs that notifications are not always making it to people.

The main library itself was already building against multiple target platforms, but the UnitTests project was not targeting all of the same ones.
In this PR I have added .NET 4.6.1 and .NET 5.0 to the `TargetFrameworks` of the UnitTests project to cover those platforms as well.
Additionally, I have added .NET 5.0 as a platform for the GitHub Action, to ensure each commit/PR will also build against that version.

Main rationales:
1. We are currently migrating some of our codebases from .NET Core 3.1 to .NET 5.0, and would feel more comfortable if the libraries we are targeting were also be targeting .NET 5.0 in their CI pipelines.

2. I looked at cleaning up some of the UnitTests code (PRs #169 and #170), and felt more comfortable in my changes if the UnitTests ran accross more, and different, platforms.

dvsekhvalnov commented:
Well, it seems that only relevant file to add .net5 testing target is `.github/workflows/build-test.yml` then?

Everything else seems to be code formatting.

dvsekhvalnov commented on UnitTests/UnitTests.csproj:
```diff
@@ -3,7 +3,7 @@
     <Description>UnitTests Class Library</Description>
     <Authors>Administrator</Authors>
-    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp3.1;net50</TargetFrameworks>
```
`UnitTests` project is netcore only tests. 
.Net Framework tests living in 2 other projects, no need to include additional targets here.

csrakowski replied:
I completely missed the other SLNs and CSprojects, so I have removed the .NET 4.6.1 reference here.
I was unable to find any other project referencing .NET 5. Was there supposed to be one, or is it okay to leave the `net50` here?

dvsekhvalnov replied:
Why we may need to build unittests for .net5.0? 

There is nothing in UnitTests that specific to net5 at the moment (no conditional compilation statements). When you run against .net50 environment it will just fine run .net3.1 build.

csrakowski replied:
Currently there are indeed no conditional compiles for .NET 5 in either the unittest project or main library, otherwise this platform would have already been there as a target.
The reason you need it here in the target frameworks is so you can actually run the unittests from the .net5 runtime, using `dotnet test jose-jwt.sln --framework=net50` in the pipeline.
See https://github.com/dvsekhvalnov/jose-jwt/pull/168/files#diff-063ca77e012f959eba648db60e6868875fd1e70e5f5ac081193d848f8d61ef32R56

dvsekhvalnov replied:
ok, if we need it, than fine.

dvsekhvalnov commented on UnitTests/UnitTests.csproj:
```diff
@@ -14,6 +14,7 @@
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
     <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
```
so we need this one at all?

csrakowski replied:
This is a "leftover" from when .net 4.6.1 was a target platform here.
If you plan on leaving the .net 4.6.1 in it's own UnitTest project, then this whole ItemGroup can actually go.
If you might want to merge all the test platforms into one test project, then having it in here will mean it's prepared for that time.

If you want, I can revert adding this line. Or remove the whole ItemGroup all together.

dvsekhvalnov replied:
No, i don't think i'm planning to keep `net461` target here. Guess it was added in effort to make single test suite, but it appeared to be pain in the ass. Too many conditional complication stuff.

csrakowski replied:
Okay, I've removed the ItemGroup to clean up the file.

dvsekhvalnov commented:
@csrakowski looks fine in general, can we remove jose-jwt/jose-jwt.csproj from commit? it seems to be only formatting changes.

dvsekhvalnov commented:
Thanks, merging now :)
